### ISSUE TITLE
ResponseWriter.println(char) does not print newline

### ISSUE THREAD

**Issue by mperktold** [eclipse/jetty.project#7031]
**Jetty version(s)**
10.0.7

**Java version/vendor** `(use: java -version)`
openjdk version "11.0.9" 2020-10-20
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.9+11)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.9+11, mixed mode)

**OS type/version**
Windows 10, 64 bit

**Description**
When a single character is passed to `response.getWriter().println`, the newline after the character is not printed. In other words, for characters, `print` and `println` have the same effect.

Other overloads for other parameter types do not seem to have this problem, just character.

**How to reproduce?**

Call a servlet that writes single characters using `response.getWriter().println(char)` and verify the output.

**Comment by joakime** [comment]
This is quite the surprising issue.

For many reasons ...

* We don't have a test case for this.
* Nor do any of the other dozen open source projects that use Jetty that we build/test releases against.
* Nor does the actual Servlet TCK have a test for this.

**Comment by joakime** [comment]
So ...

The `response.getWriter().println()` and `response.getWriter().println(char)` are both implemented by `javax.servlet.ServletOutputStream`.
Both wind up calling `javax.servlet.ServletOutputStream.println(String)` with appropriate parameters.  (this is the behaviour in Jakarta EE 8 / Jakarta Servlet 4.x, unknown if Java EE 7 / Servlet 4.x does the same, need to investigate)

The implementation of `javax.servlet.ServletOutputStream.println(String)` in Jetty is `org.eclipse.jetty.server.HttpOutput.println(String)`.

I cannot see a bug there, as even the various small test of `response.getWriter().println("X")` (String vs char) would fail the same way.

**Comment by mperktold** [comment]
I'm talking about this one in [`ResponseWriter`](https://github.com/eclipse/jetty.project/blob/b56edf511ab4399122ea2c6162a4a5988870f479/jetty-server/src/main/java/org/eclipse/jetty/server/ResponseWriter.java#L345-L366):

```java
    @Override
    public void println(char c)
    {
        try
        {
            synchronized (lock)
            {
                isOpen();
                out.write(c);
            }
        }
        catch (InterruptedIOException ex)
        {
            if (LOG.isDebugEnabled())
                LOG.debug("Write interrupted", ex);
            Thread.currentThread().interrupt();
        }
        catch (IOException ex)
        {
            setError(ex);
        }
    }
```

It only calls `out.write(c)` without writing a newline, whereas e.g. [`println(String)`](https://github.com/eclipse/jetty.project/blob/b56edf511ab4399122ea2c6162a4a5988870f479/jetty-server/src/main/java/org/eclipse/jetty/server/ResponseWriter.java#L428) calls `out.write(__lineSeparator);` after writing the string.

I actually saw this behavior in a custom `org.eclipse.jetty.server.Handler`, not a servlet. Since both receive a `HttpServletRequest`, I assumed that the behavior would be the same for servlets, but perhaps I was wrong.

**Comment by joakime** [comment]
I'm in the process of improving code coverage for both `response.getServletOutputStream()` and `response.getWriter()` based actions.
This has already revealed more issues in other places.
Stay tuned for PR.

**Comment by joakime** [comment]
Opened PR #7032

**Comment by joakime** [comment]
Two bugs were found and fixed in PR #7032

1. `response.getWriter().println((char)c)` would not include the `System.lineSeparator()`.
2. `response.getWriter().println((boolean)b)` would include `System.lineSeparator()` twice.

**Comment by joakime** [comment]
Some history, this bug was introduced in the Unchecked Print Writer commit 0dc394869b63a4573f9882275f1df59d89e2f810 (introduced in Jetty 9.3.0)

This has been fixed in all active branches.
